### PR TITLE
feat(x/precisebank): Display 0 reserve balance to module consumers

### PR DESCRIPTION
**Pull request by drklee3**
Module reserve represents fractional balances, so it should be hidden to consumers to not have a misleading total balance that doubles the fractional balances.

This modifies `GetBalance()` and `SpendableCoin()` to always return zero coins when fetching the reserve address balance.

### CODE REVIEW

File: x/precisebank/keeper/view_integration_test.go [hunk 1]
```diff
@@ -129,3 +129,30 @@ func (suite *viewIntegrationTestSuite) TestKeeper_SpendableCoin() {
...
+	// Reserve balances should not be shown to consumers of x/precisebank, as it
+	// represents the fractional balances of accounts.
+
+	moduleAddr := authtypes.NewModuleAddress(types.ModuleName)
+	addr1 := sdk.AccAddress{1}
+
+	// Make the reserve hold a non-zero balance
+	// Mint fractional coins to an account, which should cause a mint of 1
+	// integer coin to the reserve to back it.
+	suite.MintToAccount(addr1, sdk.NewCoins(sdk.NewCoin(types.ExtendedCoinDenom, sdk.NewInt(1000))))
```
Comment (pirtleshell) [inline x/precisebank/keeper/view_integration_test.go:143]: can you mint some of another denom to this address & write a test confirming that that denom is unaffected?

on mainnet, we never expect any other denom, but feels worthwhile to verify that the overridden zero balance is only for akava

Comment (pirtleshell) [reply x/precisebank/keeper/view_integration_test.go:143]: correction / update: we _do_ expect another denom. it will have ukava if the sum of all the different fractional balances is greater than the conversion factor.

so the ask: can you also mint ukava to the module in this test & verify that it correctly responds with a non-zero balance when queried?

**Review by pirtleshell — APPROVED** [review]

**Review by nddeluca — APPROVED** [review]
Nice work, great use of comments 👍
